### ISSUE TITLE
Drop unselected columns on the sstable reader level

### ISSUE THREAD

**Issue by denesb** [scylladb/scylla#6393]
Currently the sstable reader reads all cells from each row, even if it belongs to a column that is not selected in the slice. The slice is applied when the result set is built from the data. This is inefficient because we processing and keeping in memory data that will be dropped later.

Found when working on a unit test for reverse queries. I wanted to ensure I don't hit the reverse query limit, by un-selecting the large blob field from my schema, but found out this was in vain because the limit is enforced in the reader pipeline, at which point the data still contained data for all regular columns.

**Comment by slivne** [comment]
We have todo that AFAIK to ente data into the cache - the cache holds only complete rows, not partial rows.

**Comment by denesb** [comment]
Makes sense. We could implement such slicing in all concrete mutation sources: memtable, cache, sstable reader (we can use the same code in all places). Cache could then edit the slice before passing down to the sstable layer, such that it includes all columns.

Questionable, whether the benefits are worth this much trouble.
